File: Muc_Do_1_Co_Ban/Bai4_MangHieu1D/Solution.py
import sys
def solve():
    try:
        sys.stdin = open('BAI4.INP', 'r')
        sys.stdout = open('BAI4.OUT', 'w')
    except:
        pass
    data = sys.stdin.read().split()
    if not data: return
    n = int(data[0])
    q = int(data[1])
    idx = 2
    d = [0] * (n + 2)
    for _ in range(q):
        l = int(data[idx])
        r = int(data[idx+1])
        v = int(data[idx+2])
        idx += 3
        d[l] += v
        d[r+1] -= v
    cur = 0
    ans = []
    for i in range(1, n + 1):
        cur += d[i]
        ans.append(str(cur))
    print(' '.join(ans))
solve()

File: Muc_Do_1_Co_Ban/Bai4_MangHieu1D/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    if (fopen("BAI4.INP", "r")) {
        freopen("BAI4.INP", "r", stdin);
        freopen("BAI4.OUT", "w", stdout);
    }
    int n, q;
    if (!(cin >> n >> q)) return 0;
    vector<long long> d(n + 2, 0);
    while (q--) {
        int l, r;
        long long v;
        cin >> l >> r >> v;
        d[l] += v;
        d[r+1] -= v;
    }
    long long cur = 0;
    for (int i = 1; i <= n; ++i) {
        cur += d[i];
        cout << cur << (i == n ? "" : " ");
    }
    cout << "\n";
    return 0;
}
